File: frontend/src/types/caseStudy.ts
/** 案例知识库类型定义 */

export interface CaseStudy {
  id: string
  title: string
  category: string
  case_type: 'success' | 'lesson'
  description?: string
  key_metrics?: Record<string, unknown>
  tags: string[]
  takeaways: string[]
  source_url?: string
  created_at: string
  updated_at: string
}

export interface CaseStats {
  total: number
  by_category: Record<string, number>
  by_type: Record<string, number>
  top_tags: Array<{name: string; count: number}>
}

File: backend/app/schemas/case_study.py
"""案例知识库 Pydantic schemas."""

from pydantic import BaseModel
from typing import Optional


class CaseStudyCreate(BaseModel):
    title: str
    category: str
    description: Optional[str] = None
    key_metrics: Optional[dict] = None
    tags: list[str] = []
    takeaways: list[str] = []
    source_url: Optional[str] = None
    case_type: str = "success"


class CaseStudyUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    key_metrics: Optional[dict] = None
    tags: Optional[list[str]] = None
    takeaways: Optional[list[str]] = None
    source_url: Optional[str] = None
    case_type: Optional[str] = None


class CaseStudyResponse(BaseModel):
    id: str
    title: str
    category: str
    case_type: str
    description: Optional[str] = None
    key_metrics: Optional[dict] = None
    tags: list[str]
    takeaways: list[str]
    source_url: Optional[str] = None
    created_at: str
    updated_at: str


class CaseStats(BaseModel):
    total: int
    by_category: dict
    by_type: dict
    top_tags: list[dict]
